scripts/warnings_overview.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+"""
+This script analyzes a build log from Valhalla and pretty-prints warnings and their occurrences.
+"""
+
+from argparse import ArgumentParser
+from collections import Counter, defaultdict
+from pathlib import Path
+import re
+from typing import List, Optional
+
+GCC = "gcc"
+CLANG = "clang"
+MSVC = "msvc"
```

Comment:
I can take a stab a the msvc warnings once I get the build properly working.

Good idea to check in a script like this. I just copied and pasted the output from VS to a file and ran a one-off script. 😅